src/conversions/array.rs

```diff
@@ -61,8 +63,16 @@ mod min_const_generics {
     where
         T: FromPyObject<'s>,
     {
-        let seq = <crate::types::PySequence as PyTryFrom>::try_from(obj)?;
-        let seq_len = seq.len()? as usize;
+        // Types that pass `PySequence_Check` usually implement enough of the sequence protocol
+        // to support this function and if not, we will only fail extraction safely.
+        let seq = unsafe {
+            if ffi::PySequence_Check(obj.as_ptr()) != 0 {
+                <PySequence as PyTryFrom>::try_from_unchecked(obj)
+            } else {
+                return Err(PyDowncastError::new(obj, "Sequence").into());
```

Comment:
Willing to add a test for this error branch (ie. pass a non-iterable) to get the coverage to 100%? 😃

Reply:
Added a test case triggering this error path.